pkgs/development/compilers/rust/default.nix

```diff
@@ -9,29 +9,29 @@ let
 in
 rec {
   rustc = callPackage ./rustc.nix {
-    shortVersion = "1.15.1";
+    inherit llvm targets targetPatches targetToolchains rustPlatform;
+
     isRelease = true;
     forceBundledLLVM = false;
     configureFlags = [ "--release-channel=stable" ];
-    srcRev = "021bd294c039bd54aa5c4aa85bcdffb0d24bc892";
-    srcSha = "1dp7cjxj8nv960jxkq3p18agh9bpfb69ac14x284jmhwyksim3y7";
+
+    shortVersion = "1.17.0";
+    src = fetchurl {
+      url = "https://static.rust-lang.org/dist/rustc-${shortVersion}-src.tar.gz";
```

Comment:
likewise, shortVersion is not in scope here, I think the enclosing set should be `rec`